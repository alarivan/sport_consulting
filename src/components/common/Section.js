import React from "react"
import styled from "styled-components"

import Img from "gatsby-image"

import Title from "../styled/Title.css.js"
import Section from "../styled/Section.css.js"
import Content from "../styled/Content.css.js"

const Image = styled(Img)`
  display: block;
  margin: 0 auto;
  border-radius: 1em;
  padding: 0.5rem;
  max-width: 600px;
`

const Inner = styled.div`
  padding: 1rem;

  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    justify-content: center;
  }
`

const Item = styled.div``

export default ({ reverse, title, showTitle, image, children }) => {
  return (
    <Section>
      {showTitle && <Title data-testid="section-title">{title}</Title>}
      <Inner>
        <Item style={{ order: reverse + 1 }}>
          <Image fluid={image} alt="" />
        </Item>
        <Item style={{ order: !reverse + 1 }}>
          <Content>{children}</Content>
        </Item>
      </Inner>
    </Section>
  )
}
