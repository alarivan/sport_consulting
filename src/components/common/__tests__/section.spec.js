import React from "react"
import { render } from "@testing-library/react"
import Section from "../Section"

describe("Section", () => {
  it("renders correctly", () => {
    const { asFragment, queryByTestId } = render(<Section>Children</Section>)

    expect(queryByTestId("section-title")).toBeNull()
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly with title", () => {
    const props = { title: "Title", showTitle: true }
    const { asFragment, getByTestId } = render(
      <Section {...props}>Children</Section>
    )

    expect(getByTestId("section-title").innerHTML).toContain(props.title)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly with reverse modifier", () => {
    const props = { reverse: true }
    const { asFragment } = render(<Section {...props}>Children</Section>)

    expect(asFragment()).toMatchSnapshot()
  })
})
