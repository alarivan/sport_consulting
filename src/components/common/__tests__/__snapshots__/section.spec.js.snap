// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders correctly 1`] = `
<DocumentFragment>
  .c2 {
  box-sizing: border-box;
  display: block;
  padding-left: calc(0rem / 2);
  padding-right: calc(0rem / 2);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  max-width: 100%;
  display: block;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4 {
  box-sizing: border-box;
  display: block;
  padding-left: calc(0rem / 2);
  padding-right: calc(0rem / 2);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  max-width: 100%;
  display: block;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 {
  box-sizing: border-box;
  margin-left: calc(-0rem / 2);
  margin-right: calc(-0rem / 2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
}

.c0 {
  width: 100%;
  padding: 2rem 0;
}

.c5 {
  padding: 1rem 2rem;
}

.c3 {
  display: block;
  margin: 0 auto;
  border-radius: 1em;
  padding: 0.5rem;
}

@media (min-width:30rem) {
  .c2 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:48rem) {
  .c2 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:60rem) {
  .c2 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
    display: block;
  }
}

@media (min-width:30rem) {
  .c4 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:48rem) {
  .c4 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:60rem) {
  .c4 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
    display: block;
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media (min-width:30rem) {
  .c1 {
    margin-left: calc(-0rem / 2);
    margin-right: calc(-0rem / 2);
  }
}

@media (min-width:48rem) {
  .c1 {
    margin-left: calc(-0rem / 2);
    margin-right: calc(-0rem / 2);
  }
}

@media (min-width:60rem) {
  .c1 {
    margin-left: calc(-0rem / 2);
    margin-right: calc(-0rem / 2);
  }
}

<section
    class="c0"
  >
    <div
      class="c1"
      style="width: 100%;"
    >
      <div
        class="c2"
      >
        <img
          alt=""
          class="c3"
        />
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          Children
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Section renders correctly with reverse modifier 1`] = `
<DocumentFragment>
  .c2 {
  box-sizing: border-box;
  display: block;
  padding-left: calc(0rem / 2);
  padding-right: calc(0rem / 2);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  max-width: 100%;
  display: block;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4 {
  box-sizing: border-box;
  display: block;
  padding-left: calc(0rem / 2);
  padding-right: calc(0rem / 2);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  max-width: 100%;
  display: block;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 {
  box-sizing: border-box;
  margin-left: calc(-0rem / 2);
  margin-right: calc(-0rem / 2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
}

.c0 {
  width: 100%;
  padding: 2rem 0;
}

.c5 {
  padding: 1rem 2rem;
}

.c3 {
  display: block;
  margin: 0 auto;
  border-radius: 1em;
  padding: 0.5rem;
}

@media (min-width:30rem) {
  .c2 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:48rem) {
  .c2 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:60rem) {
  .c2 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
    display: block;
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media (min-width:30rem) {
  .c4 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:48rem) {
  .c4 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:60rem) {
  .c4 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
    display: block;
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }
}

@media (min-width:30rem) {
  .c1 {
    margin-left: calc(-0rem / 2);
    margin-right: calc(-0rem / 2);
  }
}

@media (min-width:48rem) {
  .c1 {
    margin-left: calc(-0rem / 2);
    margin-right: calc(-0rem / 2);
  }
}

@media (min-width:60rem) {
  .c1 {
    margin-left: calc(-0rem / 2);
    margin-right: calc(-0rem / 2);
  }
}

<section
    class="c0"
  >
    <div
      class="c1"
      style="width: 100%;"
    >
      <div
        class="c2"
      >
        <img
          alt=""
          class="c3"
        />
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          Children
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Section renders correctly with title 1`] = `
<DocumentFragment>
  .c3 {
  box-sizing: border-box;
  display: block;
  padding-left: calc(0rem / 2);
  padding-right: calc(0rem / 2);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  max-width: 100%;
  display: block;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c5 {
  box-sizing: border-box;
  display: block;
  padding-left: calc(0rem / 2);
  padding-right: calc(0rem / 2);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  max-width: 100%;
  display: block;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c2 {
  box-sizing: border-box;
  margin-left: calc(-0rem / 2);
  margin-right: calc(-0rem / 2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
}

.c1 {
  width: 100%;
  text-align: center;
  margin: 0 0 1.5rem 0;
}

.c0 {
  width: 100%;
  padding: 2rem 0;
}

.c6 {
  padding: 1rem 2rem;
}

.c4 {
  display: block;
  margin: 0 auto;
  border-radius: 1em;
  padding: 0.5rem;
}

@media (min-width:30rem) {
  .c3 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:48rem) {
  .c3 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:60rem) {
  .c3 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
    display: block;
  }
}

@media (min-width:30rem) {
  .c5 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:48rem) {
  .c5 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
  }
}

@media (min-width:60rem) {
  .c5 {
    padding-left: calc(0rem / 2);
    padding-right: calc(0rem / 2);
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
    display: block;
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media (min-width:30rem) {
  .c2 {
    margin-left: calc(-0rem / 2);
    margin-right: calc(-0rem / 2);
  }
}

@media (min-width:48rem) {
  .c2 {
    margin-left: calc(-0rem / 2);
    margin-right: calc(-0rem / 2);
  }
}

@media (min-width:60rem) {
  .c2 {
    margin-left: calc(-0rem / 2);
    margin-right: calc(-0rem / 2);
  }
}

<section
    class="c0"
  >
    <h1
      class="c1"
      data-testid="section-title"
    >
      Title
    </h1>
    <div
      class="c2"
      style="width: 100%;"
    >
      <div
        class="c3"
      >
        <img
          alt=""
          class="c4"
        />
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          Children
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
